custom SDL "breaks" autoconf

I tried to build dosbox-x in archlinux and noticed that autoconf can't find fluidsynth lib:
```
checking fluidsynth.h usability... yes
checking fluidsynth.h presence... yes
checking for fluidsynth.h... yes
checking for fluid_synth_sysex in -lfluidsynth... no
```
after investigating `config.log` i found this:
<details>
  <summary>failure log</summary>

```
configure:8752: checking for fluid_synth_sysex in -lfluidsynth
configure:8777: gcc -o conftest -g -std=gnu11  -O2 -I/home/tormoz/temp/dosbox-x/vs2015/sdlnet/linux-host/include -I/home/tormoz/temp/dosbox-x/vs2015/sdlnet/linux-host/include/SDL  -Wno-int-to-void-pointer-cast   -Wno-address-of-packed-member   -Wno-format-zero-length   -Wno-missing-field-initializers   -Wno-strict-aliasing   -Wno-implicit-fallthrough   -Wno-deprecated-declarations   -Wconversion-null   -Wsign-promo   -Wlogical-op   -pedantic   -Wunused   -Wextra   -Wall   -I/home/tormoz/temp/dosbox-x/vs2015/sdl/linux-host/include/SDL -D_GNU_SOURCE=1 -D_REENTRANT -I/home/tormoz/temp/dosbox-x/vs2015/sdlnet/linux-host/include -I/home/tormoz/temp/dosbox-x/vs2015/sdlnet/linux-host/include/SDL  -L/home/tormoz/temp/dosbox-x/vs2015/sdlnet/linux-host/lib conftest.c -lfluidsynth  -lasound -lm -ldl -lpthread -L/home/tormoz/temp/dosbox-x/vs2015/sdl/linux-host/lib -Wl,-rpath,/home/tormoz/temp/dosbox-x/vs2015/sdl/linux-host/lib -lSDL -lpthread -lm -ldl -lX11 -lXext -lts -lpthread -ltinfo >&5
cc1: warning: command line option '-Wconversion-null' is valid for C++/ObjC++ but not for C
cc1: warning: command line option '-Wsign-promo' is valid for C++/ObjC++ but not for C
cc1: warning: unrecognized command line option '-Wno-int-to-void-pointer-cast'Fluid
/usr/bin/ld: /home/tormoz/temp/dosbox-x/vs2015/sdl/linux-host/lib/libSDL.a(SDL_x11gl.o): in function `X11_GL_Shutdown':
/home/tormoz/temp/dosbox-x/vs2015/sdl/src/video/x11/SDL_x11gl.c:335: undefined reference to `glFinish'
/usr/bin/ld: /home/tormoz/temp/dosbox-x/vs2015/sdl/src/video/x11/SDL_x11gl.c:336: undefined reference to `glFlush'
/usr/bin/ld: /home/tormoz/temp/dosbox-x/vs2015/sdl/src/video/x11/SDL_x11gl.c:344: undefined reference to `glFinish'
/usr/bin/ld: /home/tormoz/temp/dosbox-x/vs2015/sdl/src/video/x11/SDL_x11gl.c:345: undefined reference to `glFlush'
/usr/bin/ld: /home/tormoz/temp/dosbox-x/vs2015/sdl/linux-host/lib/libSDL.a(SDL_x11gl.o): in function `X11_GL_MakeCurrent':
/home/tormoz/temp/dosbox-x/vs2015/sdl/src/video/x11/SDL_x11gl.c:367: undefined reference to `glFinish'
/usr/bin/ld: /home/tormoz/temp/dosbox-x/vs2015/sdl/src/video/x11/SDL_x11gl.c:368: undefined reference to `glFlush'
/usr/bin/ld: /home/tormoz/temp/dosbox-x/vs2015/sdl/src/video/x11/SDL_x11gl.c:381: undefined reference to `glFinish'
/usr/bin/ld: /home/tormoz/temp/dosbox-x/vs2015/sdl/src/video/x11/SDL_x11gl.c:382: undefined reference to `glFlush'
/usr/bin/ld: /home/tormoz/temp/dosbox-x/vs2015/sdl/linux-host/lib/libSDL.a(SDL_x11gl.o): in function `X11_GL_CreateContext':
/home/tormoz/temp/dosbox-x/vs2015/sdl/src/video/x11/SDL_x11gl.c:269: undefined reference to `glFinish'
/usr/bin/ld: /home/tormoz/temp/dosbox-x/vs2015/sdl/src/video/x11/SDL_x11gl.c:270: undefined reference to `glFlush'
/usr/bin/ld: /home/tormoz/temp/dosbox-x/vs2015/sdl/src/video/x11/SDL_x11gl.c:279: undefined reference to `glFinish'
/usr/bin/ld: /home/tormoz/temp/dosbox-x/vs2015/sdl/src/video/x11/SDL_x11gl.c:280: undefined reference to `glFlush'
collect2: error: ld returned 1 exit status
configure:8777: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "dosbox-x"
| #define PACKAGE_TARNAME "dosbox-x"
| #define PACKAGE_VERSION "0.83.2"
| #define PACKAGE_STRING "dosbox-x 0.83.2"
| #define PACKAGE_BUGREPORT "https://github.com/joncampbell123/dosbox-x/issues"
| #define PACKAGE_URL "http://dosbox-x.software"
| #define PACKAGE "dosbox-x"
| #define VERSION "0.83.2"
| #define STDC_HEADERS 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_SYS_STAT_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_STRING_H 1
| #define HAVE_MEMORY_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_INTTYPES_H 1
| #define HAVE_STDINT_H 1
| #define HAVE_UNISTD_H 1
| #define SIZEOF_UNSIGNED_CHAR 1
| #define SIZEOF_UNSIGNED_SHORT 2
| #define SIZEOF_UNSIGNED_INT 4
| #define SIZEOF_UNSIGNED_LONG 8
| #define SIZEOF_UNSIGNED_LONG_LONG 8
| #define SIZEOF_INT_P 8
| #define HAVE_STDLIB_H 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_SYS_SOCKET_H 1
| #define HAVE_NETINET_IN_H 1
| #define HAVE_PWD_H 1
| #define ENVIRON_LINKED 1
| #define DIRENT_HAS_D_TYPE 1
| #define DB_HAVE_CLOCK_GETTIME 1
| #define C_HAS_ATTRIBUTE 1
| #define C_ATTRIBUTE_ALWAYS_INLINE 1
| #define C_HAS_BUILTIN_EXPECT 1
| #define C_HAVE_MPROTECT 1
| #define HAVE_REALPATH 1
| #define C_SET_PRIORITY 1
| #define LINUX 1
| #define C_DIRECTLPT 1
| #define C_DIRECTSERIAL 1
| #define C_TARGETCPU X86_64
| #define HAVE_LIBASOUND 1
| #define HAVE_ALSA 1
| #define C_SDL1 1
| /* end confdefs.h.  */
| 
| /* Override any GCC internal prototype to avoid an error.
|    Use char because int might match the return type of a GCC
|    builtin and then its argument prototype would still apply.  */
| #ifdef __cplusplus
| extern "C"
| #endif
| char fluid_synth_sysex ();
| int
| main ()
| {
| return fluid_synth_sysex ();
|   ;
|   return 0;
| }
configure:8786: result: no
```
</details>

In fact almost any `AC_CHECK_LIB` fails. I don't know exactly why some of them work (like avcodec). Probably they import GL functions somehow. I suggest check all features not related to SDL before `LDFLAGS` override

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -8,6 +8,7 @@ Unreleased
     for some reason. (joncampbell123)
   - Fix setcap on systems with split /usr (Jookia)
   - Fix building with --disable-core-inline (Jookia)
+  - Fix building not finding fluidsynth (Jookia)
 
 2022.09.0 (0.84.3)
   - Updated FFMPEG video capture to use newer API,

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -9,6 +9,8 @@ Unreleased
   - Fix setcap on systems with split /usr (Jookia)
   - Fix building with --disable-core-inline (Jookia)
   - Fix building not finding fluidsynth (Jookia)
+  - Fix some NetBSD build issues
+    (Jookia, with patch from Nia)
 
 2022.09.0 (0.84.3)
   - Updated FFMPEG video capture to use newer API,

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -6,6 +6,7 @@ Unreleased
   - Fixed FFMPEG 5 compatibility (Jookia)
   - Fix FFMPEG crash if FFMPEG fails to write header
     for some reason. (joncampbell123)
+  - Fix setcap on systems with split /usr (Jookia)
 
 2022.09.0 (0.84.3)
   - Updated FFMPEG video capture to use newer API,

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -7,6 +7,7 @@ Unreleased
   - Fix FFMPEG crash if FFMPEG fails to write header
     for some reason. (joncampbell123)
   - Fix setcap on systems with split /usr (Jookia)
+  - Fix building with --disable-core-inline (Jookia)
 
 2022.09.0 (0.84.3)
   - Updated FFMPEG video capture to use newer API,

--- a/Makefile.am
+++ b/Makefile.am
@@ -132,11 +132,11 @@ install: src/dosbox-x
 	install -m 644 $(srcdir)/contrib/linux/dosbox-x.1 $(DESTDIR)$(mandir)/man1
 	mkdir -p $(DESTDIR)$(prefix)/share/bash-completion/completions
 	install -m 644 $(srcdir)/contrib/linux/dosbox-x $(DESTDIR)$(prefix)/share/bash-completion/completions
-	-test -x /usr/sbin/setcap && setcap cap_net_raw=ep $(DESTDIR)$(bindir)/dosbox-x
+	-command -v setcap >/dev/null && setcap cap_net_raw=ep $(DESTDIR)$(bindir)/dosbox-x || true
 
 install_strip: src/dosbox-x install
 	install -m 755 -s src/dosbox-x $(DESTDIR)$(bindir)
-	-test -x /usr/sbin/setcap && setcap cap_net_raw=ep $(DESTDIR)$(bindir)/dosbox-x
+	-command -v setcap >/dev/null && setcap cap_net_raw=ep $(DESTDIR)$(bindir)/dosbox-x || true
 
 uninstall:
 	rm -f $(DESTDIR)$(bindir)/dosbox-x

--- a/configure.ac
+++ b/configure.ac
@@ -429,6 +429,43 @@ else
   fi
 fi
 
+dnl FEATURE: Whether to use OpenGL
+AH_TEMPLATE(C_OPENGL,[Define to 1 to use opengl display output support])
+AC_ARG_ENABLE(opengl,AC_HELP_STRING([--disable-opengl],[Disable opengl support]),enable_opengl=$enableval,enable_opengl=yes)
+AC_MSG_CHECKING(whether opengl display output will be enabled)
+
+# We need to do this before any other AC_CHECK_LIB runs just to make sure
+# linking doesn't fail if SDL is using OpenGL
+if test x$enable_opengl = xyes; then
+case "$host" in
+    *-*-darwin*)
+       AC_MSG_RESULT(yes)
+       LIBS="$LIBS -framework OpenGL"
+       AC_DEFINE(C_OPENGL,1)
+       ;;
+    *)
+       pkg-config --exists gl; RES=$?
+       if test x$RES = x0; then
+         AC_MSG_RESULT(yes)
+         CFLAGS="$CFLAGS "`pkg-config gl --cflags`
+         CPPFLAGS="$CPPFLAGS "`pkg-config gl --cflags`
+         LIBS="$LIBS "`pkg-config gl --libs`
+         AC_DEFINE(C_OPENGL,1)
+       elif test x$have_gl_h = xyes -a x$have_gl_lib = xyes ; then
+         AC_MSG_RESULT(yes)
+         LIBS="$LIBS -lGL"
+         AC_DEFINE(C_OPENGL,1)
+       elif test x$have_gl_h = xyes -a x$have_opengl32_lib = xyes ; then
+         AC_MSG_RESULT(yes)
+         LIBS="$LIBS -lopengl32"
+         AC_DEFINE(C_OPENGL,1)
+       else
+         AC_MSG_RESULT(no)
+       fi
+       ;;
+esac
+fi
+
 # FIXME: Arrggh we need the WHOLE PATH
 pwd=`realpath $srcdir`
 if [[ -z "$pwd" ]]; then pwd=`pwd`; fi
@@ -586,6 +623,7 @@ case "$host" in
        WINDRES=":"
     ;;
 esac
+
 
 dnl LIBRARY TEST: ncurses
 AC_CHECK_HEADER(curses.h,have_curses_h=yes,)
@@ -1134,42 +1172,6 @@ if test x$enable_ffmpeg = xyes; then
             AC_DEFINE(C_AVCODEC,1)
         fi
     fi
-fi
-
-dnl FEATURE: Whether to use OpenGL
-AH_TEMPLATE(C_OPENGL,[Define to 1 to use opengl display output support])
-AC_ARG_ENABLE(opengl,AC_HELP_STRING([--disable-opengl],[Disable opengl support]),enable_opengl=$enableval,enable_opengl=yes)
-AC_MSG_CHECKING(whether opengl display output will be enabled)
-
-# test
-if test x$enable_opengl = xyes; then
-case "$host" in
-    *-*-darwin*)
-       AC_MSG_RESULT(yes)
-       LIBS="$LIBS -framework OpenGL"
-       AC_DEFINE(C_OPENGL,1)
-       ;;
-    *)
-       pkg-config --exists gl; RES=$?
-       if test x$RES = x0; then
-         AC_MSG_RESULT(yes)
-         CFLAGS="$CFLAGS "`pkg-config gl --cflags`
-         CPPFLAGS="$CPPFLAGS "`pkg-config gl --cflags`
-         LIBS="$LIBS "`pkg-config gl --libs`
-         AC_DEFINE(C_OPENGL,1)
-       elif test x$have_gl_h = xyes -a x$have_gl_lib = xyes ; then
-         AC_MSG_RESULT(yes)
-         LIBS="$LIBS -lGL"
-         AC_DEFINE(C_OPENGL,1)
-       elif test x$have_gl_h = xyes -a x$have_opengl32_lib = xyes ; then
-         AC_MSG_RESULT(yes)
-         LIBS="$LIBS -lopengl32"
-         AC_DEFINE(C_OPENGL,1)
-       else
-         AC_MSG_RESULT(no)
-       fi
-       ;;
-esac
 fi
 
 dnl FEATURE: Whether to use Direct3D 9 output

--- a/configure.ac
+++ b/configure.ac
@@ -1029,7 +1029,7 @@ esac
 dnl FEATURE: Whether to use libslirp, and enable userspace TCP/IP emulation
 AH_TEMPLATE(C_SLIRP, [Define to 1 to enable userspace TCP/IP emulation, requires libslirp])
 AC_ARG_ENABLE(libslirp,AC_HELP_STRING([--disable-libslirp],[Disable libslirp support]),enable_libslirp=$enableval,enable_libslirp=yes)
-if test x$enable_libslirp == xyes ; then
+if test x$enable_libslirp = xyes ; then
   case "$host" in
     *-*-cygwin* | *-*-mingw32*)
        if test x$have_slirp_h = xyes; then

--- a/configure.ac
+++ b/configure.ac
@@ -543,7 +543,7 @@ case "$host_cpu" in
     c_targetcpu="arm"
     c_unalignedmemory=yes
     ;;
-   armv6l)
+   armv6*)
     AC_DEFINE(C_TARGETCPU,ARMV4LE)
     AC_MSG_RESULT(ARMv6 Little Endian)
     c_targetcpu="arm"

--- a/src/cpu/core_normal.cpp
+++ b/src/cpu/core_normal.cpp
@@ -51,7 +51,6 @@ extern bool ignore_opcode_63;
 #define SaveMd(off,val)	mem_writed(off,val)
 #define SaveMq(off,val) {mem_writed(off,val&0xffffffff);mem_writed(off+4,(val>>32)&0xffffffff);}
 #else 
-#include "paging.h"
 #define LoadMb(off) mem_readb_inline(off)
 #define LoadMw(off) mem_readw_inline(off)
 #define LoadMd(off) mem_readd_inline(off)

--- a/src/cpu/core_normal_286.cpp
+++ b/src/cpu/core_normal_286.cpp
@@ -21,6 +21,7 @@
 #include "cpu.h"
 #include "lazyflags.h"
 #include "callback.h"
+#include "paging.h"
 #include "pic.h"
 #include "fpu.h"
 
@@ -63,7 +64,6 @@ extern bool mustCompleteInstruction;
 #define SaveMd(off,val)	mem_writed(off,val)
 #define SaveMq(off,val) {mem_writed(off,val&0xffffffff);mem_writed(off+4,(val>>32)&0xffffffff);}
 #else 
-#include "paging.h"
 #define LoadMb(off) mem_readb_inline(off)
 #define LoadMw(off) mem_readw_inline(off)
 #define LoadMd(off) mem_readd_inline(off)

--- a/src/cpu/core_normal_8086.cpp
+++ b/src/cpu/core_normal_8086.cpp
@@ -19,6 +19,7 @@
 #include "cpu.h"
 #include "lazyflags.h"
 #include "callback.h"
+#include "paging.h"
 #include "pic.h"
 #include "fpu.h"
 
@@ -74,8 +75,6 @@ static void SaveMw(Bitu off,Bitu val) {
 #define SaveMd(off,val)	mem_writed(off,val)
 
 #else 
-
-#include "paging.h"
 
 #define LoadMb(off) mem_readb_inline(off)
 

--- a/src/cpu/core_prefetch.cpp
+++ b/src/cpu/core_prefetch.cpp
@@ -22,6 +22,7 @@
 #include "cpu.h"
 #include "lazyflags.h"
 #include "callback.h"
+#include "paging.h"
 #include "pic.h"
 #include "fpu.h"
 
@@ -49,7 +50,6 @@ extern bool ignore_opcode_63;
 #define SaveMd(off,val)	mem_writed(off,val)
 #define SaveMq(off,val) {mem_writed(off,val&0xffffffff);mem_writed(off+4,(val>>32)&0xffffffff);}
 #else 
-#include "paging.h"
 #define LoadMb(off) mem_readb_inline(off)
 #define LoadMw(off) mem_readw_inline(off)
 #define LoadMd(off) mem_readd_inline(off)

--- a/src/cpu/core_prefetch_286.cpp
+++ b/src/cpu/core_prefetch_286.cpp
@@ -22,6 +22,7 @@
 #include "cpu.h"
 #include "lazyflags.h"
 #include "callback.h"
+#include "paging.h"
 #include "pic.h"
 #include "fpu.h"
 
@@ -65,7 +66,6 @@ extern bool ignore_opcode_63;
 #define SaveMd(off,val)	mem_writed(off,val)
 #define SaveMq(off,val) {mem_writed(off,val&0xffffffff);mem_writed(off+4,(val>>32)&0xffffffff);}
 #else 
-#include "paging.h"
 #define LoadMb(off) mem_readb_inline(off)
 #define LoadMw(off) mem_readw_inline(off)
 #define LoadMd(off) mem_readd_inline(off)

--- a/src/cpu/core_prefetch_8086.cpp
+++ b/src/cpu/core_prefetch_8086.cpp
@@ -22,6 +22,7 @@
 #include "cpu.h"
 #include "lazyflags.h"
 #include "callback.h"
+#include "paging.h"
 #include "pic.h"
 #include "fpu.h"
 
@@ -78,8 +79,6 @@ static void SaveMw(Bitu off,Bitu val) {
 #define SaveMd(off,val)	mem_writed(off,val)
 
 #else 
-
-#include "paging.h"
 
 #define LoadMb(off) mem_readb_inline(off)
 

--- a/src/dos/dos_programs.cpp
+++ b/src/dos/dos_programs.cpp
@@ -6935,11 +6935,7 @@ bool AUTOTYPE::ReadDoubleArg(const std::string &name,
 	if (cmd->FindString(flag, str_value, true)) {
 		// Can the user's value be parsed?
 		const double user_value = to_finite<double>(str_value);
-#if defined(__FreeBSD__) || defined(MACOSX) || defined(EMSCRIPTEN) || ((defined(ANDROID) || defined(__ANDROID__)) && defined(__clang__))
-		if (isfinite(user_value)) { /* *sigh* Really, clang, really? */
-#else
-		if (std::isfinite(user_value)) {
-#endif
+		if (isfinite(user_value)) {
 			result = true;
 
 			// Clamp the user's value if needed

--- a/src/libs/physfs/physfs_platform_unix.c
+++ b/src/libs/physfs/physfs_platform_unix.c
@@ -53,10 +53,14 @@
 #include <sys/mnttab.h>
 #endif
 
-#ifdef PHYSFS_PLATFORM_FREEBSD
+#if defined(PHYSFS_PLATFORM_FREEBSD) || defined(__NetBSD__)
 #include <sys/sysctl.h>
 #endif
 
+#ifdef __NetBSD__
+#include <sys/statvfs.h>
+#define statfs statvfs
+#endif
 
 #include "physfs_internal.h"
 
@@ -258,6 +262,14 @@ char *__PHYSFS_platformCalcBaseDir(const char *argv0)
         char fullpath[PATH_MAX];
         size_t buflen = sizeof (fullpath);
         int mib[4] = { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1 };
+        if (sysctl(mib, 4, fullpath, &buflen, NULL, 0) != -1)
+            retval = __PHYSFS_strdup(fullpath);
+    }
+    #elif defined(__NetBSD__)
+    {
+        char fullpath[PATH_MAX];
+        size_t buflen = sizeof (fullpath);
+        int mib[4] = { CTL_KERN, KERN_PROC_ARGS, -1, KERN_PROC_PATHNAME };
         if (sysctl(mib, 4, fullpath, &buflen, NULL, 0) != -1)
             retval = __PHYSFS_strdup(fullpath);
     }

